src/order/filter/germ.lean

```diff
@@ -0,0 +1,534 @@
+  f =ᶠ[l] g :=
+h₂.mp $ h₁.mono $ λ x, le_antisymm
+
+/-- Setoid used to define the space of germs. -/
+def germ_setoid (l : filter α) (β : Type*) : setoid (α → β) :=
+{ r := eventually_eq l,
+  iseqv := ⟨eventually_eq.refl _, λ _ _, eventually_eq.symm, λ _ _ _, eventually_eq.trans⟩ }
+
+/-- The space of germs of functions `α → β` at a filter `l`. -/
+def germ (l : filter α) (β : Type*) : Type* := quotient (germ_setoid l β)
+
+namespace germ
+
+instance : has_coe_t (α → β) (germ l β) := ⟨quotient.mk'⟩
+
+instance : has_lift_t β (germ l β) := ⟨λ c, ↑(λ (x : α), c)⟩
+
+@[simp] lemma quot_mk_eq_coe (l : filter α) (f : α → β) : quot.mk _ f = (f : germ l β) := rfl
+
+@[simp] lemma mk'_eq_coe (l : filter α) (f : α → β) : quotient.mk' f = (f : germ l β) := rfl
+
+@[elab_as_eliminator]
+lemma induction_on (f : germ l β) {p : germ l β → Prop} (h : ∀ f : α → β, p f) : p f :=
+quotient.induction_on' f h
+
+@[elab_as_eliminator]
+lemma induction_on₂ (f : germ l β) (g : germ l γ) {p : germ l β → germ l γ → Prop}
+  (h : ∀ (f : α → β) (g : α → γ), p f g) : p f g :=
+quotient.induction_on₂' f g h
+
+@[elab_as_eliminator]
+lemma induction_on₃ (f : germ l β) (g : germ l γ) (h : germ l δ)
+  {p : germ l β → germ l γ → germ l δ → Prop}
+  (H : ∀ (f : α → β) (g : α → γ) (h : α → δ), p f g h) :
+  p f g h :=
+quotient.induction_on₃' f g h H
+
+/-- Given a map `F : (α → β) → (γ → δ)` that sends functions eventually equal at `l` to functions
+eventually equal at `lc`, returns a map from `germ l β` to `germ lc δ`. -/
+def map' {lc : filter γ} (F : (α → β) → (γ → δ)) (hF : (l.eventually_eq ⇒ lc.eventually_eq) F F) :
+  germ l β → germ lc δ :=
+quotient.map' F hF
+
+/-- A version of `quotient.lift_on'` using coercion instead of `quotient.mk'`. -/
+def lift_on {γ : Sort*} (f : germ l β) (F : (α → β) → γ) (hF : (l.eventually_eq ⇒ (=)) F F) : γ :=
+quotient.lift_on' f F hF
+
+@[simp] lemma map'_coe {lc : filter γ} (F : (α → β) → (γ → δ))
+  (hF : (l.eventually_eq ⇒ lc.eventually_eq) F F) (f : α → β) :
+  map' F hF f = F f :=
+rfl
+
+@[simp, norm_cast] lemma coe_eq : (f : germ l β) = g ↔ (f =ᶠ[l] g) := quotient.eq'
+
+alias coe_eq ↔ _ filter.eventually_eq.germ_eq
+
+/-- Lift a function `β → γ` to a function `germ l β → germ l γ`. -/
+def map (op : β → γ) : germ l β → germ l γ :=
+map' ((∘) op) $ λ f g H, H.mono $ λ x H, congr_arg op H
+
+@[simp] lemma map_coe (op : β → γ) (f : α → β) : map op (f : germ l β) = op ∘ f := rfl
+
+@[simp] lemma map_id : map id = (id : germ l β → germ l β) := by { ext ⟨f⟩, refl }
+
+lemma map_map (op₁ : γ → δ) (op₂ : β → γ) (f : germ l β) :
+  map op₁ (map op₂ f) = map (op₁ ∘ op₂) f :=
+induction_on f $ λ f, rfl
+
+/-- Lift a binary function `β → γ → δ` to a function `germ l β → germ l γ → germ l δ`. -/
+def map₂ (op : β → γ → δ) : germ l β → germ l γ → germ l δ :=
+quotient.map₂' (λ f g x, op (f x) (g x)) $ λ f f' Hf g g' Hg,
+Hg.mp $ Hf.mono $ λ x Hf Hg, by simp only [Hf, Hg]
+
+/-- Lift of `tendsto` to the space of germs. -/
```

Comment:
I think that a more explicit description would be better. Maybe something like: a germ at `l` of maps from `α` to `β` tends to `fb : filter β` if it is represented by a map which tends to `fb` along `l`.